Tasks/JenkinsQueueJob/util.ts

```diff
@@ -115,40 +115,92 @@ function createRootJob(queueUri: string, jobQueue: JobQueue, taskOptions: TaskOp
     return defer.promise;
 }
 
-export function pollSubmitJob(initialPostData: any, taskOptions: TaskOptions): Q.Promise<string> {
+export function pollSubmitJob(taskOptions: TaskOptions): Q.Promise<string> {
     var defer: Q.Deferred<string> = Q.defer<string>();
 
     var poll = async () => {
-        await submitJob(initialPostData, taskOptions).then((queueUri: string) => {
-            if (queueUri != null) {
-                defer.resolve(queueUri);
+        await getCrumb(taskOptions).then(async (crumb: string) => {
+            if (crumb != null) {
+                await submitJob(taskOptions).then((queueUri: string) => {
+                    if (queueUri != null) {
+                        defer.resolve(queueUri);
+                    } else {
+                        // no queueUri yet, but no failure either, so keep trying
+                        setTimeout(poll, taskOptions.pollIntervalMillis);
+                    }
+                }).fail((err: any) => {
+                    defer.reject(err);
+                });
             } else {
-                // no queueUri yet, but no failure either, so keep trying
+                // no crumb yet, but no failure either, so keep trying
                 setTimeout(poll, taskOptions.pollIntervalMillis);
             }
-        }).fail((err: any) => {
-            defer.reject(err);
-        })
+        });
     };
 
     poll();
 
     return defer.promise;
 }
 
-function submitJob(initialPostData: any, taskOptions: TaskOptions): Q.Promise<string> {
+function submitJob(taskOptions: TaskOptions): Q.Promise<string> {
     var defer: Q.Deferred<string> = Q.defer<string>();
-    tl.debug('submitJob(): ' + JSON.stringify(initialPostData));
+    tl.debug('submitJob(): ' + JSON.stringify(taskOptions));
 
-    request.post(initialPostData, function requestCallback(err, httpResponse, body) {
-        tl.debug('submitJob().requestCallback()');
+    function addCrumb(postData: any): any {
+        if (taskOptions.crumb && taskOptions.crumb != taskOptions.NO_CRUMB) {
+            postData.headers = {
+                'Jenkins-Crumb': taskOptions.crumb
```

Comment:
You probably shouldn't hardcode the crumb header name. The name is provided in the response to the crumbIssuer request. to